it/src/test/scala/integration/LoadExampleIntSpec.scala

```diff
@@ -4,43 +4,48 @@ import base.KafkaSpecBase
 import cats.data.NonEmptyList
 import cats.effect.{IO, Ref}
 import fs2.kafka.*
-import io.github.embeddedkafka.EmbeddedKafkaConfig
 import load.LoadExample
 import org.scalatest.Assertion
 import org.typelevel.log4cats.LoggerFactory
 import org.typelevel.log4cats.slf4j.Slf4jFactory
+import utils.KafkaContainer.KafkaConfig
+import utils.KafkaTestContainer
 
 import scala.concurrent.duration.*
 
-final class LoadExampleIntSpec extends KafkaSpecBase[IO] {
+final class LoadExampleIntSpec extends KafkaSpecBase[IO], KafkaTestContainer[IO] {
 
   "LoadExample" should {
-    "load previously seen messages into the store" in withKafkaContext { ctx =>
-      import ctx.{*, given}
-
-      for {
-        _      <- publishStringMessage(inputTopic, "key1", "value1")
-        _      <- runAppAndDiscard
-        _      <- publishStringMessage(inputTopic, "key2", "value2")
-        result <- runApp
-      } yield result should contain theSameElementsInOrderAs List("value1", "value2")
+    "load previously seen messages into the store" in withRunningKafka { implicit kafkaConfig =>
+      testContext { ctx =>
+        import ctx.*
+
+        for {
+          _      <- publishStringMessage(inputTopic, "key1", "value1")
+          _      <- runAppAndDiscard
+          _      <- publishStringMessage(inputTopic, "key2", "value2")
+          result <- runApp
+        } yield result should contain theSameElementsInOrderAs List("value1", "value2")
+      }
     }
 
-    "not publish previously committed messages" in withKafkaContext { ctx =>
-      import ctx.{*, given}
-
-      for {
-        _      <- publishStringMessage(inputTopic, "key1", "value1")
-        _      <- runAppAndDiscard
-        _      <- consumeStringMessage(outputTopic, autoCommit = true)
-        _      <- publishStringMessage(inputTopic, "key2", "value2")
-        _      <- runAppAndDiscard
-        result <- consumeStringMessage(outputTopic, autoCommit = true)
-      } yield result shouldBe "value2"
+    "not publish previously committed messages" in withRunningKafka { implicit kafkaConfig =>
+      testContext { ctx =>
+        import ctx.*
+
+        for {
+          _      <- publishStringMessage(inputTopic, "key1", "value1")
+          _      <- runAppAndDiscard
+          _      <- consumeStringMessage(outputTopic, autoCommit = true)
+          _      <- publishStringMessage(inputTopic, "key2", "value2")
+          _      <- runAppAndDiscard
+          result <- consumeStringMessage(outputTopic, autoCommit = true)
+        } yield result shouldBe "value2"
+      }
     }
   }
 
-  private trait TestContext {
+  final case class TestContext()(using kafkaConfig: KafkaConfig) {
```

Comment:
```suggestion
  final case class TestContext()(using KafkaConfig) {
```
as the below consumerSettings and producerSettings where it is used, names it for internal scope